File: backend/schemas/room.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class RoomCreate(BaseModel):
    nombre: str

class RoomJoin(BaseModel):
    codigo_invitacion: str

class RoomResponse(BaseModel):
    id: int
    nombre: str
    codigo_invitacion: str
    admin_id: int
    created_at: datetime

    class Config:
        from_attributes = True

class MemberResponse(BaseModel):
    user_id: int
    nombre: str
    foto_url: Optional[str] = None
    es_admin: bool
    joined_at: datetime

File: predix-frontend/types/room.ts
export interface Room {
  id: number;
  nombre: string;
  descripcion?: string;
  codigo_invitacion: string;
  admin_id: number;
  max_members?: number;
  created_at: string;
}

export interface RoomMember {
  user_id: number;
  nombre: string;
  foto_url: string | null;
  es_admin: boolean;
  joined_at: string;
}

export interface CreateRoomPayload {
  nombre: string;
}

export interface JoinRoomPayload {
  codigo_invitacion: string;
}
